### PR TITLE
Add "mainscene" and "map" commands

### PR DESCRIPTION
**Pull request by mon1kasenpai**
added a basic command that exits the player to whatever is set as the game's default scene. In my case this is the main menu because I originally wrote this for my own game and i thought it would be a good addition to the main branch. One thing i could probably add once i have a bit more time is an export var that actually lets the user select what map the disconnect command loads.

I'll probably write a better system for this down the road but ive been spending the past little while porting useful source commands for debugging and the like so ill be sure to port those over if you want :)

### CODE REVIEW

**Comment by raub** [comment]
Hello. Thank you for the proposal. In my opinion, `disconnect` is too specific as it implies a "connection" exists. I would prefer to use a different name, such as `exit` that happens to duplicate `quit` in HL. Or something else (`reload, restart, etc.`).

In general, I don't want to pollute the namespace with commands that make assumptions about the game structure and logic. For example, I don't change scenes at all (as in `change_scene*`) - I only add/remove child nodes as necessary. Although, I tried, and it somewhat works even for my scenario, so we can add this command.

Have you tested the final code? `get_tree().change_scene_to_file("application/run/main_scene")` - I initially wondered how Godot is so smart to know this is a project setting... And it isn't.

```
get_tree().change_scene_to_file(
	ProjectSettings.get_setting("application/run/main_scene")
)
```

this worked for me, however.

To sum up the changes: name, `get_setting`. Then we can merge this.

---

Regarding the other source/goldsrc commands, as per this plugin, the development focus is following:

* implement scripting logic: `alias`, `wait`, `;`, `exec`;
* consider allowing the `+/-` commands if that is going to work at all;
* think about some generic commands/cvars that are relevant for ALL projects;
* add components: top console plaque (Q1, Serious Sam), log overlay, bottom-left console (TES Oblivion/Skyrim);
* expand component options: colors, verbosity/log levels - maybe also add cvars that control those on runtime.
* improve autocompletion/matching and hints.

So, if you have more proposals, we can try to fit them in. PRs/issues are welcome!

**Comment by mon1kasenpai** [comment]
hey man, thanks for getting back to me.

I totally intend to rename the command and i probably should have done that before committing (i just copied and pasted it from my own game.) my bad, apologies. 

i did test the final code to load the main scene map and it seemed to work for me in 4.2.1. what version of godot are you running? I'll replace this bit of code in my next commit with an export var the user can set just in case they use a map that's not the main menu (a-la cruelty squad.) and it should avoid the issue all together.

I was going to write a few other general commands such as one for loading maps and spawning in scenes: would it be worth adding to this pull request or do you want me to create a new one?

**Comment by mon1kasenpai** [comment]
Alright ive updated the command name and the method used to call the main scene. i tested it in my own project and it seems to work as intended. I also added a `GsomConsole.toggle()` at the end just in case you want the console to close apon the main menu being loaded. feel free to remove or change as needed.

 Its almost midnight here and im still recovering from a really bad cold so thats all ill be doing tonight but over the next few days when im not bogged down by work and uni ill try to add what i can.

**Comment by raub** [comment]
Sure, thanks for your contribution. I hope you get better soon! I will tweak some minor things and then merge.
